compiler/rustc_typeck/src/check/fallback.rs

```diff
@@ -373,6 +407,15 @@ impl<'tcx> FnCtxt<'_, 'tcx> {
                     let a_vid = self.root_vid(a)?;
                     let b_vid = self.root_vid(b)?;
                     Some((a_vid, b_vid))
+                } else if let ty::PredicateKind::Subtype(ty::SubtypePredicate {
+                    a_is_expected: _,
+                    a,
+                    b,
+                }) = atom
+                {
+                    let a_vid = self.root_vid(a)?;
+                    let b_vid = self.root_vid(b)?;
+                    Some((a_vid, b_vid))
```

Comment:
I think this needs a comment...

Is the idea that even though it's not a "coercion", it still counts because we effectively unify them?